data/scenarios/Challenges/ice-cream.yaml

```diff
@@ -0,0 +1,242 @@
+        Let's hope this order is simple.
+    condition: |
+      customer <- robotnamed "customer";
+      as customer {has "cone"};
+  - goal:
+      - |
+        `give` the customer exactly what they request.
+      - |
+        Perhaps there is something else around the shop that can assist with
+        preparing the order...
+    condition: |
+      as base {
+        has "bitcoin";
+      };
+robots:
+  - name: base
+    dir: [0, 1]
+    devices:
+      - branch predictor
+      - treads
+      - antenna
+      - comparator
+      - workbench
+      - grabber
+      - dictionary
+      - lodestone
+      - lambda
+      - logger
+      - welder
+      - hearing aid
+      - net
+      - strange loop
+    inventory:
+      - [5000, egg]
+      - [5000, milk]
+      - [5000, sugar]
+      - [5000, ice]
+      - [5000, vanilla]
+  - name: customer
+    system: true
+    dir: [0, -1]
+    display:
+      invisible: false
+      attr: 'iron'
+    inventory:
+      - [1, bitcoin]
+    program: |
+      run "scenarios/Challenges/_ice-cream/customer.sw"
+solution: |
+   run "scenarios/Challenges/_ice-cream/solution.sw"
+entities:
+  - name: scoop
+    display:
+      char: '@'
+    description:
+      - A single scoop of vanilla ice cream
+    properties: [portable]
+  - name: cherry
+    display:
+      char: '6'
+      attr: 'red'
+    properties: [portable]
+    description:
+      - A traditional ice cream garnish
+  - name: Counter
+    display:
+      char: '▒'
+      attr: 'sand'
+    properties: [portable, known]
+    capabilities: [count]
+    description:
+      - Where customers are served.
+      - Also... it can count things.
+  - name: cone
+    display:
+      char: 'V'
+      attr: 'sand'
+    properties: [portable]
+    description:
+      - A conical, edible container for ice cream
+  - name: egg
+    display:
+      char: 'e'
+      attr: 'gold'
+    properties: [portable]
+    description:
+      - Organic from local, free-range, meticulously pampered chickens.
+  - name: milk
+    display:
+      char: 'm'
+      attr: 'silver'
+    properties: [portable]
+    description:
+      - Pasteurized, homogenized, commoditized, metabolized.
+  - name: ice
+    display:
+      char: 'i'
+      attr: 'snow'
+    properties: [portable]
+    description:
+      - Ice cream is better cold, don't you think?
+  - name: sugar
+    display:
+      char: 's'
+      attr: 'snow'
+    properties: [portable]
+    description:
+      - Basis of most desserts
+  - name: vanilla
+    display:
+      char: 'v'
+      attr: 'wood'
+    properties: [portable]
+    description:
+      - Traditional ice cream flavoring
+  - name: briefcase
+    display:
+      char: 'B'
+      attr: rubber
+    capabilities: [sum, prod]
+    properties: [portable]
+    description:
+      - Standard business accoutrement.
+      - Particularly useful for `meet`-ings.
+      - Enables the `case` expression.
```

Comment:
Oh, of course, I heard about the counter from @byorgey here, so I did not expect anything else and used `meetAll`. :smile: